### PR TITLE
GEN-644 - feat(OfferPresenter): display discount

### PR DESCRIPTION
**Pull request by guilhermespopolin**
## Describe your changes

* Update Offer Presenter so discounts get shown
* General refactoring: move `useDiscountExplanation` into global utility

image

## Justify why they are needed

* Requested by design.

### CODE REVIEW

File: apps/store/src/components/ProductPage/PurchaseForm/OfferPresenter.tsx [hunk 1]
```diff
@@ -111,28 +116,39 @@ export const OfferPresenter = (props: Props) => {
     loadingAddToCart || updateCancellationInfo.loading || updateStartDateResult.loading
 
   const discountTooltipProps = useMemo(() => {
```
Comment (alebedev) [inline apps/store/src/components/ProductPage/PurchaseForm/OfferPresenter.tsx:118]: This becomes quite long, do you think extracting named hook would improve readability?

**Review by alebedev — APPROVED** [review]
Looks good!

File: apps/store/src/components/ProductPage/PurchaseForm/DiscountTooltip/DiscountTooltip.tsx [hunk 2]
```diff
@@ -13,15 +13,11 @@ export const DiscountTooltip = ({ children, subtitle, color = 'green' }: Props)
   return (
     <Root>
       <Bubble>
-        <Text size="xs" color={color === 'green' ? 'textGreen' : 'textPrimary'} align="center">
+        <Text size="xs" color="textPrimary" align="center">
```
Comment (robinandeer) [inline apps/store/src/components/ProductPage/PurchaseForm/DiscountTooltip/DiscountTooltip.tsx:16]: Don't we support the color option anymore? 🤔

File: apps/store/src/components/ProductPage/PurchaseForm/OfferPresenter.tsx [hunk 3]
```diff
@@ -301,6 +284,56 @@ type GetCancellationOptionParams = {
...
+) => {
+  const { t } = useTranslation(['purchase-form', 'cart'])
+  const formatter = useFormatter()
+  const getDiscountExplanation = useGetDiscountExplanation()
+
+  const tooltipProps = useMemo(() => {
+    if (selectedOffer.priceMatch) {
+      const company = selectedOffer.priceMatch.externalInsurer.displayName
+
+      if (selectedOffer.priceMatch.priceReduction.amount < 1) {
```
Comment (robinandeer) [inline apps/store/src/components/ProductPage/PurchaseForm/OfferPresenter.tsx:299]: nit: `<= 0` I think reads a bit better

File: apps/store/src/components/ProductPage/PurchaseForm/DiscountTooltip/DiscountTooltip.tsx (hunk 2 above)
Comment (guilhermespopolin) [reply apps/store/src/components/ProductPage/PurchaseForm/DiscountTooltip/DiscountTooltip.tsx:16]: So based on the [new designs](https://www.figma.com/file/qUhLjrKl98PAzHov9ilaDH/Hedvig-UI-Kit?node-id=3731%3A24532&mode=dev) for discount tooltip, `color` should only change bg color.

**Review by robinandeer — APPROVED** [review]
Great stuff!

**Comment by guilhermespopolin** [comment]
Have updated this by hiding the changes under a feature flag.

**Review by robinandeer — APPROVED** [review]

**Comment by guilhermespopolin** [comment]
@guilhermespopolin merged this pull request with [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2669).

**Pull request merged**